lib/broccoli/ember-app.js

```diff
@@ -1084,12 +1178,12 @@ EmberApp.prototype.appAndDependencies = function() {
     overwrite: true
   }));
 
-  var external        = this._processedExternalTree();
   var preprocessedApp = preprocessJs(app, '/', this.name, {
     registry: this.registry
   });
 
   var postprocessedApp = this.addonPostprocessTree('js', preprocessedApp);
+  var external         = this._processedExternalTree();
```

Comment:
Moving this is not valid without us going through the work of specifying every hook's invocation order. That this needs to move is something that I'm familiar with because I want to do that for: https://github.com/ef4/ember-browserify/pull/100